src/vm/database/marf.rs

```diff
@@ -0,0 +1,90 @@
+use vm::database::{KeyValueStorage, KeyType};
+use chainstate::stacks::index::marf::MARF;
+use chainstate::stacks::index::{MARFValue, Error as MarfError};
+use chainstate::stacks::index::storage::{TrieFileStorage};
+use chainstate::burn::BlockHeaderHash;
+use util::hash::{to_hex, Sha256Sum};
+
+/// The MarfedKV struct is used to wrap a MARF data structure and side-storage
+///   for use as a K/V store for ClarityDB or the AnalysisDB.
+/// The Clarity VM and type checker do not "know" to begin/commit the block they are currently processing:
+///   each instantiation of the VM simply executes one transaction. So the block handling
+///   loop will need to invoke these two methods (begin + commit) outside of the context of the VM.
+///   NOTE: Clarity will panic if you try to execute it from a non-initialized MarfedKV context.
+///   (See: vm::tests::with_marfed_environment()) 
+pub struct MarfedKV {
+    marf: MARF,
+    // Since the MARF only stores 32 bytes of value,
+    //   we need another storage
+    side_store: Box<KeyValueStorage>
+}
+
+fn value_hash(value: &str) -> KeyType {
+    let Sha256Sum(hash_data) = Sha256Sum::from_data(value.as_bytes());
```

Comment:
I highly recommend using `TrieHash` here -- it's considerably faster than and also safer than `Sha256` (under the hood, it's SHA512/256 -- it's faster on 64-bit hardware and is immune to length-extension attacks).

Reply:
Sure -- updated to use MARFValue.from_data